java/src/main/native/src/ScalarJni.cpp

```diff
@@ -138,6 +138,22 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_Scalar_getListAsColumnView(JNIEnv *e
   CATCH_STD(env, 0);
 }
 
+JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_Scalar_getChildrenFromStructScalar(JNIEnv *env, jclass,
+                                                                                    jlong scalar_handle) {
+  JNI_NULL_CHECK(env, scalar_handle, "scalar handle is null", 0);
+  try {
+    cudf::jni::auto_set_device(env);
+    const auto s                  = reinterpret_cast<cudf::struct_scalar*>(scalar_handle);
+    const cudf::table_view& table = s->view();
+    cudf::jni::native_jlongArray column_handles(env, table.num_columns());
```

Comment:
nit. There is a native class just for this situation. `native_jpointerArray`. You should be able to just assign the pointers to it then call `get_jArray` to return it. No `reinterpret_cast` needed.

Reply:
Fixed.